### PR TITLE
VSH/UI: Add VSH to game grid

### PR DESCRIPTION
**Pull request by elad335**
* Custom VSH config.
* VSH shortcut creation.

image

### CODE REVIEW

**Comment by ArbestRi** [comment]
VSH image should be changed because the icon is literally PlayStation Home.

**Comment by Darkhost1999** [comment]
If you got a better icon just right click-> custom icon. And replace it.

File: rpcs3/rpcs3qt/emu_settings.cpp [hunk 1]
```diff
@@ -90,7 +91,7 @@ bool emu_settings::Init()
 	return true;
 }
 
-void emu_settings::LoadSettings(const std::string& title_id)
+void emu_settings::LoadSettings(const std::string& title_id, const std::string& game_path)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/emu_settings.cpp:94]: the game_path seems irrelevant as you already pass the self name as seríal to get_custom_config_path in other places

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 2]
```diff
@@ -677,12 +696,14 @@ void game_list_frame::Refresh(const bool from_drive, const bool scroll_after)
 
 			m_mutex_cat.unlock();
 
+			const std::string app_identifier = game.path.starts_with(dev_flash) ? game.path.substr(game.path.find_last_of(fs::delim) + 1) : game.serial;
+
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/game_list_frame.cpp:700]: this seems irrelevant as game.serial is already the exact same

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 3]
```diff
@@ -580,46 +581,64 @@ void game_list_frame::Refresh(const bool from_drive, const bool scroll_after)
...
-			game.category     = std::string(psf::get_string(psf, "CATEGORY", cat_unknown));
-			game.fw           = std::string(psf::get_string(psf, "PS3_SYSTEM_VER", cat_unknown_localized));
-			game.parental_lvl = psf::get_integer(psf, "PARENTAL_LEVEL", 0);
-			game.resolution   = psf::get_integer(psf, "RESOLUTION", 0);
-			game.sound_format = psf::get_integer(psf, "SOUND_FORMAT", 0);
-			game.bootable     = psf::get_integer(psf, "BOOTABLE", 0);
-			game.attr         = psf::get_integer(psf, "ATTRIBUTE", 0);
-
-			if (m_show_custom_icons)
+			if (m_show_custom_icons && game.icon_path.empty())
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/game_list_frame.cpp:636]: this logic breaks the custom icon for vsh.

File: rpcs3/Emu/System.cpp [hunk 4]
```diff
@@ -1191,35 +1193,33 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
...
-					{
-						sys_log.fatal("Failed to apply custom config: %s", config_path);
-					}
-				}
-
-				// Load custom config-2
-				if (fs::file cfg_file{ m_path + ".yml" })
+					from_dev_flash ? rpcs3::utils::get_custom_config_path(m_path.substr(m_path.find_last_of(fs::delim) + 1)) : std::string{},
+					m_path + ".yml",
+					config_path
```
[the remark this answers is not in the document]
Comment (Megamouse) [reply rpcs3/Emu/System.cpp:1201]: on second thought, doesn't get_custom_config_path already work with m_title_id anyway?

[the remark this answers is not in the document]
Comment (elad335) [reply rpcs3/Emu/System.cpp:1201]: I rather not modify m_title_id or any other emulation variables for now, so it is empty.

[the remark this answers is not in the document]
Comment (Megamouse) [reply rpcs3/Emu/System.cpp:1201]: ah. I was looking at the m_title_id of a different class

File: rpcs3/rpcs3qt/game_list_frame.cpp (hunk 3 above)
Comment (elad335) [reply rpcs3/rpcs3qt/game_list_frame.cpp]: Fixed.

**Comment by Megamouse** [comment]
I think the name could be more user friendly. vsh.self isn't very descriptive for the average user.
As a new user I'd wonder what that's supposed to be

File: rpcs3/rpcs3qt/game_list_frame.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/rpcs3qt/game_list_frame.cpp:699]: just tested and it definitely is

File: rpcs3/Emu/System.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (Megamouse) [reply rpcs3/Emu/System.cpp:1201]: the suggested change still removes an obsolete call to get_custom_config_path

[the remark this answers is not in the document]
Comment (elad335) [reply rpcs3/Emu/System.cpp]: Fixed.

File: rpcs3/rpcs3qt/game_list_frame.cpp (hunk 2 above)
Comment (elad335) [reply rpcs3/rpcs3qt/game_list_frame.cpp]: Fixed.

**Comment by elad335** [comment]
Added.

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 5]
```diff
@@ -580,48 +581,84 @@ void game_list_frame::Refresh(const bool from_drive, const bool scroll_after)
...
+					if (usz pos = path_vfs.find_first_not_of(fs::delim); pos != umax && pos != 0)
+					{
+						path_vfs = path_vfs.substr(pos);
+					}
+
+					if (Localized().title.titles.contains(path_vfs))
+					{
+						game.name = Localized().title.titles.at(path_vfs).toStdString();
+					}
+				}
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/game_list_frame.cpp:633]: I don't understand why you're adding this complicated logic when this whole segment is tailored for vsh anyway.

Comment (elad335) [reply rpcs3/rpcs3qt/game_list_frame.cpp:633]: There are more stuff in dev_flash other than VSH, this is for them as well.

**Review by Megamouse — APPROVED** [review]

**Comment by MSuih** [comment]
I still think we should use different icon than the one for PS Home. Maybe a screenshot of the background would be enough?

**Comment by elad335** [comment]
it can be revised later. It needs to consider rpcs3 package size, offline availability etc, so dev_flash stuff has been the easy choice.
